src/hugr/validate.rs

```diff
@@ -162,13 +162,19 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
         // Check operation-specific constraints. Firstly that type args are correct
         // (Good to call `resolve_extension_ops` immediately before this
         //   - see https://github.com/CQCL-DEV/hugr/issues/508 )
-        if let OpType::LeafOp(crate::ops::LeafOp::CustomOp(b)) = op_type {
-            for arg in b.args() {
+        match op_type {
```

Comment:
Beginnings of a new trait here (currently we have a trait+method for `validate_op_children` but this is not about children)